sdk/sdk-skeleton.go

```diff
@@ -0,0 +1,126 @@
+import "time"
+
+type Asset struct{}
+
+//// Methods directly on the SDK
+
+func NewChannel() {
+
+}
+
+func NewChannelListenerService() {
+
+}
+
+func GetChannel() {
+
+}
+
+//// Channel State Machine
+
+type ChannelStateMachine struct {
+	ChannelID              string
+	Status                 string // waiting | active | closing | closed
+	Asset                  Asset
+	InitiatorEscrowAccount string
+	ResponderClaimAmount   int
+	ReserveEscrowAccount   string
+	InitiatorClaimAmount   int
+	IsInitiator            bool
+	MyAccount              string
+	MyClaimAMount          int
+	OtherAccount           string
+	OtherClaimAmount       int
+}
+
+func (c *ChannelStateMachine) Init() {
+	return nil
+}
+
+func (c *ChannelStateMachine) RegisterMonitoringNotificationHandler(handler MonitoringNotificationHandler) error {
+	return nil
+}
+
+func (c *ChannelStateMachine) CheckChannel() (*ChannelCheckResponse, error) {
+	return nil, error
+}
+
+func (c *ChannelStateMachine) Payment(sendAmount int, timeoutMillis int) error {
+	return nil
+}
+
+func (c *ChannelStateMachine) CloseStart(declareID int) error {
+	return nil
+}
+
+func (c *ChannelStateMachine) CloseCoordinated(timeoutMillis int, id string) (newStatus string, err error) {
+	return "", nil
+}
+
+func (c *ChannelStateMachine) CloseUncoordinated(id string) error {
+	return nil
+}
+
+func (c *ChannelStateMachine) GetLatestDeclarationTx() (*TxInfo, error) {
+	return nil, nil
+}
+
+func (c *ChannelStateMachine) GetLatestCloseTx(id string) (*TxInfo, error) {
+	return nil, nil
+}
+
+type TxInfo struct {
+	ID               string
+	Type             string // declaration | close
+	Seq              int64
+	InitiatorBalance int
+	ResponderBalance int
```

Comment:
The transactions won't capture balances, but rather amount owing to each other which will only be a portion of the balances. @nikhilsaraf What's the goal of these two fields? I'm wondering if we need them, or need something else.

Reply:
`PayoutAmount`?

Reply:
`BalanceOwing` is maybe another way to think about it, because the amount in the payment won't be the total payout amount, but rather just what doesn't already exist in the recipients account.

Reply:
Yes, agreed. I was thinking this would be better as a function instead of as a variable, mostly for convenience to the SDK user so they can just call MyBalance() rather than doing the calculations themselves using the internals of this struct.

Reply:
moved to helper method:
https://github.com/stellar/experimental-payment-channels/pull/25/commits/5bd4f54a1d3e26e434e2f21521191fde56c108d9